gremlin-python/src/main/jython/gremlin_python/structure/io/graphsonV2d0.py

```diff
@@ -339,18 +339,17 @@ class DateIO(_GraphSONTypeIO):
     python_type = datetime.datetime
     graphson_type = "g:Date"
     graphson_base_type = "Date"
+    epoch = datetime.datetime(1970, 1, 1)
 
     @classmethod
     def dictify(cls, obj, writer):
-        # Java timestamp expects miliseconds
+        # Java timestamp expects milliseconds.
         if six.PY3:
-            pts = obj.timestamp()
+            pts = (obj - cls.epoch) / datetime.timedelta(seconds=1)
```

Comment:
To me this is a bit of a confusing way to get to a timestamp. I might suggest having a look at datetime handling in the Cassandra driver: https://github.com/datastax/python-driver/blob/master/cassandra/cqltypes.py#L577-L579
This technique has a few advantages:
- No special cases per python runtime
- Handles datetime inputs with timezones attached, normalizing back to UTC
- More readable (imo)